bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/admin/ContentAdminResource.java

```diff
@@ -0,0 +1,155 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.core.bind.jaxrs.admin;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.commons.lang.StringUtils;
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.bind.jaxrs.IndyResources;
+import org.commonjava.indy.bind.jaxrs.util.REST;
+import org.commonjava.indy.bind.jaxrs.util.ResponseHelper;
+import org.commonjava.indy.core.bind.jaxrs.util.TrackingContentController;
+import org.commonjava.indy.core.model.TrackedContent;
+import org.commonjava.indy.core.model.TrackedContentEntry;
+import org.commonjava.indy.core.model.TrackingKey;
+import org.commonjava.indy.core.model.dto.ContentTransferDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriInfo;
+import java.io.File;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.commonjava.indy.util.ApplicationContent.application_zip;
+
+@Api( value = "Tracking Record Admin Content Access", description = "Manages Content tracking records." )
+@Path( "/api/admin/content" )
+@ApplicationScoped
+@REST
+public class ContentAdminResource
+                implements IndyResources
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private TrackingContentController controller;
+
+    @Inject
+    private ResponseHelper responseHelper;
+
+    // TODO need to disable folo {id}/repo/zip endpoint
+    @ApiOperation(
+                    "Retrieve the content referenced in a tracking record as a ZIP-compressed Maven repository directory." )
+    @ApiResponses( { @ApiResponse( code = 200, response = File.class, message = "ZIP repository content" ),
+                    @ApiResponse( code = 404, message = "No such tracking record" ) } )
+    @Path( "/repo/zip" )
+    @POST
+    @Produces( application_zip )
+    public File getZipRepository( @Context final UriInfo uriInfo, final TrackedContent record )
```

Comment:
I'm thinking if we should keep the TrackedContent and TrackedContentEntry in the content service model, because it is folo-specific model. We should also move them out of the content-service. 
I think this getZipREpository only need storeKey and path for transfer handling, so maybe we should use an alternative DTO model like:
{ 
 key: $trackingKey, 
 content:[
  { 
     storeKey: $storeKey,
     path: xxxxxx
  },
  .........
 ] 
}

Reply:
updated to use a different version of object for `getZipRepository`.

I'm not sure where to move the folo classes. Do you have any suggestions?